core/async/src/messaging.rs

```diff
@@ -206,6 +210,115 @@ impl<T: Debug, R> Debug for MessageWithCallback<T, R> {
     }
 }
 
+pub struct SpanWrappedMsg<T> {
```

Comment:
FWIW I believe it'd be more fitting to make this part of o11y crate as this is more of a o11y thing than a message thing really.

Reply:
Yeah, moved it to `o11y`.